Save target report images

iche033 opened the pull request:
Signed-off-by: Ian Chen <ichen@osrfoundation.org>

The changes enable 2 sets of images to be saved when a target is reported:
1. An image from the image stream transmitted by the user (over comms) is saved.
    * I added a new dependency, `cv_bridge`, for saving `sensor_msgs::msgs::Image` to file. So you'll need to run `rosdep` again or `apt install ros-galactic-cv-bridge` if you don't have this package.
    * This image file is saved  to `~/.ros/mbzirc`. Reason is that the video relay node is run on the bridge container on the cloud, and in this container we are uploading the `.ros` directory to s3. 
    * Only the last image of a particular target type is saved, e.g. if the user reported small target vessel twice, the last one will be saved.
3. An image is saved by the `GameLogicPlugin`. The image is taken from the rendering camera in the simulator when a report is received. This is the original uncompressed image. 
    * The image is saved to `/tmp/ign/mbzirc/logs/target_images`.
    * One image is saved per report, e.g. f the user reported small target vessel twice, there will be 2 images.

To test, follow the same instructions for reporting targets in https://github.com/osrf/mbzirc/pull/195. Make sure to apply the diff for making the drone and vessel static. After each target report, you should see new png files saved in `~/.ros/mbzirc/` and `/tmp/ign/mbzirc/logs/target_images`.

caguero reviewed (approved):
Works like a charm!

My only minor suggestion is that if the only purpose for the new mutex member is to protect `this->saveImage`, we could change this member to be `std::atomic<bool>`.

iche033 commented:
> My only minor suggestion is that if the only purpose for the new mutex member is to protect `this->saveImage`, we could change this member to be `std::atomic<bool>`.

yeah I did that first but later on I added `saveImagePrefix` which also needs to be protected so I changed to use a mutex
